src/prognostic_variables.jl

```diff
@@ -96,3 +96,26 @@ function Base.zeros(::Type{PrognosticVariables{NF}},
     pres = zeros(SurfaceLeapfrog{NF},_lmax,_mmax)                     # 2 leapfrog time steps for pres
     return PrognosticVariables(layers,pres,lmax,mmax,N_LEAPFROG,nlev)
 end
+
+
+# GPU methods
+function Adapt.adapt_structure(to, pvl::PrognosticVariablesLayer)
+    PrognosticVariablesLayer(Adapt.adapt(to, pvl.vor), 
+                             Adapt.adapt(to, pvl.div), 
+                             Adapt.adapt(to, pvl.temp),
+                             Adapt.adapt(to, pvl.humid))
+end
+
+function Adapt.adapt_structure(to, p_leap::PrognosticVariablesLeapfrog)
+    PrognosticVariablesLeapfrog(Adapt.adapt(to, p_leap.leapfrog))
+end
+
+function Adapt.adapt_structure(to, s_leap::SurfaceLeapfrog)
+    SurfaceLeapfrog(Adapt.adapt(to, s_leap.leapfrog))
+end
+
+function Adapt.adapt_structure(to , pv::PrognosticVariables)
+    PrognosticVariables(Adapt.adapt(to, pv.layers),
+                        Adapt.adapt(to, pv.pres)
```

Comment:
`,` missing here

Reply:
added with [9e9a452](https://github.com/milankl/SpeedyWeather.jl/pull/143/commits/9e9a452ea9dcbd68ab998a827a598baf91f6c2d1)

Reply:
Huh, I am surprised that this didn't yield an error with the test in `test/kernelabstractions.jl`. Without GPU CI it can't test if transferring everything to GPU works, but it should yield errors for syntax errors. I'll have to look into this.